### PR TITLE
fish-shell: update to 3.1.0

### PR DESCRIPTION
**Pull request by yamorain**

### CODE REVIEW

File: srcpkgs/fish-shell/template [hunk 1]
```diff
@@ -1,7 +1,7 @@
 # Template file for 'fish-shell'
 pkgname=fish-shell
-version=3.0.2
-revision=2
+version=3.1.0
+revision=3
```
Comment (sgn) [inline srcpkgs/fish-shell/template]: revision=1?

File: srcpkgs/fish-shell/template [hunk 2]
```diff
@@ -11,7 +11,7 @@ maintainer="Nathan Owens <ndowens04@gmail.com>"
 license="GPL-2.0-only"
 homepage="https://fishshell.com/"
 distfiles="https://github.com/fish-shell/fish-shell/releases/download/${version}/fish-${version}.tar.gz"
-checksum=14728ccc6b8e053d01526ebbd0822ca4eb0235e6487e832ec1d0d22f1395430e
+checksum=e5db1e6839685c56f172e1000c138e290add4aa521f187df4cd79d4eab294368
 register_shell="/bin/fish /usr/bin/fish"
```
Comment (sgn) [inline srcpkgs/fish-shell/template]:
```
if [ "$XBPS_TARGET_NO_ATOMIC8" ]; then
        makedepends+=" libatomic-devel"
fi
```

**Comment by Johnnynator** [comment]
This currently conflicts with the ripgrep packages, both provide the same completion in the same dir. Idk why we don't install fish completions in non fish packages to ` /usr/share/fish/vendor_completions.d`. Pls add a commit that removes the one from the ripgrep package.

```
xbps-query -o /usr/share/fish/completions/rg.fish 
fish-shell-3.1b1_1: /usr/share/fish/completions/rg.fish (regular file)
ripgrep-11.0.2_1: /usr/share/fish/completions/rg.fish (regular file)
```

**Comment by Chocimier** [comment]
Fish do not depend on bc since 3.0, please remove.

**Comment by Chocimier** [comment]
As content of ripgrep changes, revision needs to be increased

**Pull request merged**